src/ucp/core/ucp_request.inl

```diff
@@ -70,6 +70,9 @@
         /* will never put it into mpool */ \
         uint32_t _flags = ((_req)->flags |= UCP_REQUEST_FLAG_COMPLETED); \
         (_req)->status = (_status); \
+        \
+        ucp_request_id_check(_req, ==, UCP_REQUEST_ID_INVALID); \
```

Comment:
maybe we should move this to ucp_request_put? instead of ucp_request_id_reset()?

Reply:
done
having it here and in `ucp_request_put()` - will cover all cases

Reply:
ok

Reply:
should not we also just check id in `ucp_request_put` instead of resetting?

Reply:
> should not we also just check id in `ucp_request_put` instead of resetting?

we should
done